### PR TITLE
Commited first time in the 48541_activity branch

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

File: app/assets/javascripts/activities.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (shakil-fg) [inline app/assets/javascripts/activities.coffee]: Remove empty file

File: app/assets/stylesheets/activities.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the activities controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (shakil-fg) [inline app/assets/stylesheets/activities.scss]: Remove empty file

File: app/helpers/activities_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module ActivitiesHelper
+end
```
Comment (shakil-fg) [inline app/helpers/activities_helper.rb]: Remove empty file

File: app/models/subject.rb [hunk 4]
```diff
@@ -11,4 +11,9 @@ class Subject < ActiveRecord::Base
         allow_destroy: true
 
   validates :name, presence: true
-end
\ No newline at end of file
+
+  # enum key: ["subject.create", "subject.update"]
+  # scope :activity_by_subject, -> {where(key: "subject.create").order("created_at desc")}
```
Comment (shakil-fg) [inline app/models/subject.rb]: Remove temporary comments

File: app/views/supervisor/subjects/index.html.erb [hunk 5]
```diff
@@ -17,7 +16,12 @@
       </div>
     <% end %>
   </div>
-  <div class="col-sm-1">
-    <!-- PLACEHOLDER: Progress -->
+  <div class=".col-sm-6 pull-right">
```
Comment (shakil-fg) [inline app/views/supervisor/subjects/index.html.erb]: Fix inconsistent div size (bootstrap)

**Comment by manazirahsan** [comment]
Okey

**Comment by shakil-fg** [comment]
OK

File: app/controllers/supervisor/subjects_controller.rb [hunk 6]
```diff
@@ -60,4 +61,8 @@ def subject_params
     params.require(:subject).permit(:name, :description, tasks_attributes:
       [:id, :name, :description, :_destroy])
   end
+
+  def load_activities
+    @activities = PublicActivity::Activity.my_activity("subject.create")
```
Comment (MuhammadTamzid) [inline app/controllers/supervisor/subjects_controller.rb]: remove ()

File: app/controllers/supervisor/subjects_controller.rb [hunk 7]
```diff
@@ -60,4 +61,8 @@ def subject_params
     params.require(:subject).permit(:name, :description, tasks_attributes:
```
Comment (MuhammadTamzid) [inline app/controllers/supervisor/subjects_controller.rb]: remove ()

**Comment by manazirahsan** [comment]
okey

File: config/routes.rb [hunk 8]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
 
+  get 'activities/index'
```
Comment (dieunb) [inline config/routes.rb]: double quote pls

File: config/routes.rb [hunk 9]
```diff
@@ -8,6 +10,7 @@
     get 'about'=> 'static_pages#about'
     get 'contact'=> 'static_pages#contact'
 
+    resources :activities
```
Comment (dieunb) [inline config/routes.rb]: may duplicated with above define

**Comment by noor-framgia** [comment]
ok

File: app/views/shared/_activity.html.erb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+ <% @activities.each do |activity| %>
+        <%= link_to activity.owner.name, supervisor_user_path(activity.owner) if activity.owner %>
```
Comment (dieunb) [inline app/views/shared/_activity.html.erb]: indent

File: app/views/shared/_activity.html.erb [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+        <%= t :created_text %> <%= link_to activity.trackable.name, supervisor_subject_path(activity.trackable) if activity.trackable %>
+     <%= t :at_text %>&nbsp<%= activity.created_at.to_s[0..-4] if activity.created_at %>
```
Comment (dieunb) [inline app/views/shared/_activity.html.erb]: indent

**Comment by manazirahsan** [comment]
okey

**Comment by noor-framgia** [comment]
okey

**Pull request merged**